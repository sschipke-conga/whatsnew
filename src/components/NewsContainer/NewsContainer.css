main {
grid-area: news-container;
display: grid;
grid-template-columns: repeat(2, 1fr);
overflow:scroll;
}