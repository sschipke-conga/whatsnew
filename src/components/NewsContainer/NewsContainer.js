import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

const NewsContainer = ({articles}) => {
  let newsArticles = articles.map((story,i) => (
    <NewsArticle key={i} story={story} />
  ))
  return <main>{newsArticles}</main>
}

export default NewsContainer;