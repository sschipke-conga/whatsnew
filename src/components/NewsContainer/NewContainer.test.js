import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer'

describe('NewsArticle', () => {
  let stories = [{ img: '//www.pic.com', headline: 'EXTRA EXTRA', description: 'Read all about it' }, { img: '//www.pictures.com', headline: 'Fake news', description: 'don\'t read it'}]
  it('should match a snap shot', () => {
    const wrapper = shallow(<NewsContainer
      articles={stories}
    />)
    expect(wrapper).toMatchSnapshot();
  })
});