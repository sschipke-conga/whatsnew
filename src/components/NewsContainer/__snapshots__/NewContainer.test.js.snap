// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match a snap shot 1`] = `
<main>
  <NewsArticle
    key="0"
    story={
      Object {
        "description": "Read all about it",
        "headline": "EXTRA EXTRA",
        "img": "//www.pic.com",
      }
    }
  />
  <NewsArticle
    key="1"
    story={
      Object {
        "description": "don't read it",
        "headline": "Fake news",
        "img": "//www.pictures.com",
      }
    }
  />
</main>
`;
